packages/web-react/src/components/ControlButton/useControlButtonProps.ts

```diff
@@ -0,0 +1,34 @@
+import { type ClickEvent, type SpiritControlButtonProps } from '../../types';
+
+const handleClick = (event: ClickEvent, isDisabled?: boolean, onClick?: (event: ClickEvent) => void) => {
+  if (isDisabled) {
+    event.preventDefault();
+
+    return;
+  }
+
+  if (onClick) {
+    onClick(event);
+  }
+};
```

Comment:
Uses even Button and ButtonLink. Separate into shared _hook_, if we have any?

Reply:
I refactored the `handleClick` function to a shared hook.